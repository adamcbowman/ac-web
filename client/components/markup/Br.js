import React from 'react'
import PropTypes from 'prop-types'
import { memo } from 'utils/react'
import styles from './Br.css'

Br.propTypes = {
    ribbon: PropTypes.bool,
}

function Br({ ribbon }) {
    const className = ribbon ? 'Ribbon' : 'Main'

    return <div className={styles[className]} />
}

export default memo.static(Br)
