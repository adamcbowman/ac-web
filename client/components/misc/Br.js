import React from 'react'
import PropTypes from 'prop-types'
import CSSModules from 'react-css-modules'
import styles from './Br.css'

Br.propTypes = {
    ribbon: PropTypes.bool,
}

function Br({ribbon = false}) {
    return (
        <div styleName={ribbon ? 'Ribbon' : 'Main'}></div>
    )
}

export default CSSModules(Br, styles)
