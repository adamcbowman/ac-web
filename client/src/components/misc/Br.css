@import "../../styles/colors.css";

.Main {
    composes: import Hash from "../../styles/Textures.css";
    display: block;
    height: 8px;
}

.Ribbon {
    composes: Main;

    &:before {
        content: '';
        display: inline-block;
        float: left;
        background-color: var(--brand-primary);
        width: 50px;
        height: 100%;
    }
}
