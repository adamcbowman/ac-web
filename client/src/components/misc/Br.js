import React, {PropTypes} from 'react'
import CSSModules from 'react-css-modules'
import styles from './Br.css'

Br.propTypes = {
    withRibbon: PropTypes.bool,
}

function Br({withRibbon = false}) {
    return (
        <div styleName={withRibbon ? 'Ribbon' : 'Main'}></div>
    )
}

export default CSSModules(Br, styles)
