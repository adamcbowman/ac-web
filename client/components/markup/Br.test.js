import React from 'react'
import Renderer from 'react-test-renderer'
import Br from './Br'

test('br component', () => {
    expect(Renderer.create(<Br />)).toMatchSnapshot()
    expect(Renderer.create(<Br ribbon />)).toMatchSnapshot()
})
