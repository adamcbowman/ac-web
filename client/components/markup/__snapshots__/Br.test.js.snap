// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`br component 1`] = `
<div
  className="Main"
/>
`;

exports[`br component 2`] = `
<div
  className="Ribbon"
/>
`;
